planner/core/preprocess.go

```diff
@@ -767,6 +776,13 @@ func checkIndexInfo(indexName string, IndexPartSpecifications []*ast.IndexPartSp
 	return checkDuplicateColumnName(IndexPartSpecifications)
 }
 
+// checkEmptyIndexInfo checks  empty index name
+func checkEmptyIndexInfo(name string, isEmptyIndex bool) error {
+	if isEmptyIndex {
+		return ddl.ErrWrongNameForIndex.GenWithStackByArgs(name)
+	}
+	return nil
+}
```

Comment:
This function is simple enough. Please move the body to `checkCreateIndexGrammar`.